js/src/block-editor/components/edit.js

```diff
@@ -1,37 +1,50 @@
 /**
  * WordPress dependencies
  */
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/components';
 import ServerSideRender from '@wordpress/server-side-render';
 
 /**
  * Internal dependencies
  */
-import { GcbInspector, FormControls } from './';
-import icons from '../../../../assets/icons.json';
+import { Fields, GcbInspector } from './';
+import { getIconComponent } from '../../common/helpers';
 
 /**
- * The Edit function for the block.
+ * The editor component for the block.
  *
  * @param {Object} props The props of this component.
  * @param {Object} props.blockProps The block's props.
  * @param {Object} props.block The block.
- * @return {Function} The Edit function for the block.
+ * @return {React.ReactElement} The editor display.
  */
 const Edit = ( { blockProps, block } ) => {
 	const { attributes, className, isSelected } = blockProps;
 
-	if ( 'undefined' === typeof icons[ block.icon ] ) {
-		icons[ block.icon ] = '';
-	}
-
 	return (
 		<>
 			<GcbInspector blockProps={ blockProps } block={ block } />
 			<div className={ className } key={ `form-controls-${ block.name }` } >
 				{ isSelected ? (
 					<div className="block-form">
-						<h3 dangerouslySetInnerHTML={ { __html: icons[ block.icon ] + ' ' + block.title } } />
-						<FormControls blockProps={ blockProps } block={ block } />
```

Comment:
`<FormControls>` was just a wrapper for `<Fields>`, so this simply uses `<Fields>`